app/src/main/java/com/duckduckgo/app/trackerdetection/TrackerDetector.kt

```diff
@@ -16,25 +16,26 @@
 
 package com.duckduckgo.app.trackerdetection
 
-import com.duckduckgo.app.trackerdetection.TrackerDetectionClient.ClientName
+import com.duckduckgo.app.trackerdetection.Client.ClientName
+import com.duckduckgo.app.trackerdetection.model.ResourceType
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class TrackerDetector @Inject constructor() {
 
-    private val clients = ArrayList<TrackerDetectionClient>()
+    private val clients = ArrayList<Client>()
 
-    fun addClient(client: TrackerDetectionClient) {
+    fun addClient(client: Client) {
         clients.add(client)
     }
 
     fun hasClient(name: ClientName): Boolean {
-        return clients.filter { it.name == name }.isNotEmpty()
+        return clients.any { it.name == name }
     }
 
     fun shouldBlock(url: String, documentUrl: String, resourceType: ResourceType): Boolean {
-        for (client: TrackerDetectionClient in clients) {
+        for (client: Client in clients) {
```

Comment:
why not use `clients.any` as you've done above?

Reply:
Good question, probably just kotlin newbie conservatism. There is a lot more happening here and I thought it was more obvious as is. But I'll take the plunge and update this too!

Reply:
I don't mind which approach you go for, but feel like we use the same approach in the same class.